### PR TITLE
Move hardhat and hardhat-ethers to dev dependencies

### PR DESCRIPTION
**Pull request by zhongeric**
From what I can tell, `hardhat` and `@nomiclabs/hardhat-ethers` are two packages that I added when adding integ tests to SOR that should be in devDependencies. Not sure if this affects the .lock file change for interface but I can test locally.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -31,7 +31,6 @@
     "node": ">=10"
   },
   "dependencies": {
-    "@nomiclabs/hardhat-ethers": "^2.0.6",
     "@types/async-retry": "^1.4.2",
```
Comment (zzmp) [inline package.json:34]: The entire `@types` scope should probably also be moved to devDependencies - they are only used for compilation.

File: package.json [hunk 2]
```diff
@@ -53,7 +52,6 @@
     "ethers": "^5.6.1",
```
Comment (zzmp) [inline package.json:52]: Is bunyan-debug-stream development-only?

**Review by zzmp — APPROVED** [review]

**Comment by zzmp** [comment]
This will likely affect the lockfile for interface. If you'd like, I can own updating interface once this is merged and released under a new patch version.

**Review by willpote — APPROVED** [review]
Agree with Zacks comments but approved

File: package.json (hunk 2 above)
Comment (zhongeric) [reply package.json:52]: Yeah it's just. for logging

**Comment by zhongeric** [comment]
> This will likely affect the lockfile for interface. If you'd like, I can own updating interface once this is merged and released under a new patch version.

That would be great as I might not be here anymore after it's merged haha

**Comment by zhongeric** [comment]
Moved all @types into devDep, had to bump @types/async-retry to 1.4.4 (what it was resolving to anyways) since 1.4.2 was throwing errors when building.

**Comment by zzmp** [comment]
> > This will likely affect the lockfile for interface. If you'd like, I can own updating interface once this is merged and released under a new patch version.
> 
> That would be great as I might not be here anymore after it's merged haha

This is actually taken care of in https://github.com/Uniswap/interface/pull/4326.

**Pull request merged**